### PR TITLE
LPD-24202 Adapt the REST Context path for modifiable system objects that are part of the root model

### PR DESCRIPTION
**Pull request by magjed4289**
https://issues.liferay.com/browse/LPD-24202

### CODE REVIEW

**Comment by magjed4289** [comment]
ci:test:relevant

File: modules/apps/object/object-service/src/main/java/com/liferay/object/model/impl/ObjectDefinitionImpl.java [hunk 1]
```diff
@@ -20,10 +21,14 @@
...
 import java.util.Objects;
 
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
 /**
  * @author Marco Leo
  * @author Brian Wing Shun Chan
  */
+@Component(service = ObjectDefinition.class)
```
Comment (ccorreagg) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/model/impl/ObjectDefinitionImpl.java:31]: We cannot do this... The ObjectDefinition can not be registered as an OSGi component... What is the purpose of that? Should we just have one ObjectDefinition in the whole portal? It is the implementation of the ObjectDefinition database DTO

File: modules/apps/object/object-service/src/main/java/com/liferay/object/model/impl/ObjectDefinitionImpl.java [hunk 2]
```diff
@@ -122,9 +127,26 @@ public String getRESTContextPath() {
 			throw new UnsupportedOperationException();
 		}
 
+		ObjectDefinition rootObjectDefinition =
+			_objectDefinitionPersistence.fetchByPrimaryKey(
```
Comment (ccorreagg) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/model/impl/ObjectDefinitionImpl.java:131]: In the master's branch, I see this code to do the same action:

```
ObjectDefinition rootObjectDefinition =
			ObjectDefinitionLocalServiceUtil.fetchObjectDefinition(
				getRootObjectDefinitionId());
```

Comment (ccorreagg) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/model/impl/ObjectDefinitionImpl.java:131]: This is a similar pattern I found to mock the `ObjectDefinitionLocalServiceUtil`:

image

**Review by ccorreagg — CHANGES_REQUESTED** [review]
Please, review my comments. We can not register a DTO implementation as a Component

File: modules/apps/object/object-service/src/main/java/com/liferay/object/model/impl/ObjectDefinitionImpl.java (hunk 1 above)
Comment (magjed4289) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/model/impl/ObjectDefinitionImpl.java:31]: SF doesn't allow me to put @Reference without @Component, I added it cause it was easier for me than changing persistence to ObjectDefinitionUtil method in tests (it is implemented as static)

Comment (magjed4289) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/model/impl/ObjectDefinitionImpl.java:31]: I reread your comment, and yeah, it is quite not what we're trying to achieve xD Honestly, I did not even think what @\Component tag implies, SF told me to add it, so I did 😅

File: modules/apps/object/object-service/src/test/java/com/liferay/object/model/impl/ObjectDefinitionImplTest.java [hunk 3]
```diff
@@ -32,76 +34,124 @@ public class ObjectDefinitionImplTest {
...
-			});
+				});
 
-		if (rootObjectDefinitionName != null) {
-			ObjectDefinition rootObjectDefinition = new ObjectDefinitionImpl();
+			if (rootObjectDefinitionName != null) {
+				ObjectDefinition rootObjectDefinition =
+					new ObjectDefinitionImpl();
 
-			rootObjectDefinition.setModifiable(true);
```
Comment (magjed4289) [inline modules/apps/object/object-service/src/test/java/com/liferay/object/model/impl/ObjectDefinitionImplTest.java]: I removed setting modifiable and system properties for root object as we don't need them for testing this method

**Comment by ccorreagg** [comment]
ci:forward

**Comment by magjed4289** [comment]
Test failures are not related to the PR.

**Comment by ccorreagg** [comment]
ci:forward:force

**Pull request closed without merging**